sdk/typescript/test/e2e/dynamic-fields.test.ts

```diff
@@ -51,18 +53,17 @@ describe.each([{ useLocalTxnBuilder: true }])(
       expect(dynamic_fields.nextCursor).not.toEqual(null);
     });
     it('go to next cursor', async () => {
-      const dynamic_fields = await toolbox.provider.getDynamicFields(
-        parent_objectID,
-        null,
-        1
-      );
-      const dynamic_fields2 = await toolbox.provider.getDynamicFields(
-        parent_objectID,
-        dynamic_fields.nextCursor,
-        null
-      );
-      expect(dynamic_fields2.data.length).greaterThan(0);
-      expect(dynamic_fields.nextCursor).not.toEqual(null);
+      return await toolbox.provider
+        .getDynamicFields(parent_objectID, null, 1)
+        .then(async function (dynamic_fields) {
+          expect(dynamic_fields.nextCursor).not.toEqual(null);
+
+          return await toolbox.provider
+            .getDynamicFields(parent_objectID, dynamic_fields.nextCursor, null)
+            .then(function (dynamic_fields2) {
+              expect(dynamic_fields2.data.length).greaterThanOrEqual(0);
+            });
+        });
```

Comment:
I actually don't think the refactoring of mixing await with callback makes sense. The bug seems to be in the backend rather than here